src/main/java/teammates/ui/controller/InstructorFeedbackEditPageData.java

```diff
@@ -12,68 +12,297 @@
+                                                     getPreviewAsStudentOptions(studentList), 
+                                                     getPreviewAsInstructorOptions(instructorList));
+    }
+
+
+    private void buildExistingQuestionForm(String feedbackSessionName,
+                                           int questionsSize,
+                                           Map<String, Boolean> questionHasResponses,
+                                           String courseId, FeedbackQuestionAttributes question) {
+        FeedbackQuestionEditForm qnForm = new FeedbackQuestionEditForm();
+        qnForm.setAction(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACK_QUESTION_EDIT));
+        qnForm.setCourseId(courseId);
+        
+        qnForm.setFeedbackSessionName(feedbackSessionName);
+        qnForm.setQuestion(question);
+        qnForm.setQuestionNumberSuffix("-" + question.questionNumber);
+        
+        qnForm.setNumOfQuestionsOnPage(questionsSize);
+        qnForm.setQuestionNumberOptions(getQuestionNumberOptions(questionsSize));
+        
+        FeedbackQuestionFeedbackPathSettings feedbackPathSettings = new FeedbackQuestionFeedbackPathSettings();
+        FeedbackQuestionVisibilitySettings visibilitySettings = new FeedbackQuestionVisibilitySettings();
+        qnForm.setFeedbackPathSettings(feedbackPathSettings);
+        qnForm.setVisibilitySettings(visibilitySettings);
+        feedbackPathSettings.setGiverParticipantOptions(getParticipantOptions(question, true));
+        feedbackPathSettings.setRecipientParticipantOptions(getParticipantOptions(question, false));
+        
+        
+        // maps for setting visibility
+        Map<String, Boolean> isGiverNameVisible = new HashMap<String, Boolean>();
+        for (FeedbackParticipantType giverType : question.showGiverNameTo) {
+            isGiverNameVisible.put(giverType.name(), true);
+        }
+        visibilitySettings.setGiverNameVisibleFor(isGiverNameVisible);
+        
+        Map<String, Boolean> isRecipientNameVisible = new HashMap<String, Boolean>();
+        for (FeedbackParticipantType recipientType : question.showRecipientNameTo) {
+            isRecipientNameVisible.put(recipientType.name(), true);
+        }
+        visibilitySettings.setRecipientNameVisibleFor(isRecipientNameVisible);
+        
+        Map<String, Boolean> isResponsesVisible = new HashMap<String, Boolean>();
+        for (FeedbackParticipantType participantType : question.showResponsesTo) {
+            isResponsesVisible.put(participantType.name(), true);
+        }
+        visibilitySettings.setResponseVisibleFor(isResponsesVisible);
+        
+        feedbackPathSettings.setNumberOfEntitiesToGiveFeedbackToChecked(question.numberOfEntitiesToGiveFeedbackTo == Const.MAX_POSSIBLE_RECIPIENTS);
+        feedbackPathSettings.setNumOfEntitiesToGiveFeedbackToValue(question.numberOfEntitiesToGiveFeedbackTo == Const.MAX_POSSIBLE_RECIPIENTS ?
```

Comment:
Maybe extract that into a boolean?
Or you could use the `is*` method on `feedbackPathSettings` for the 2nd line to show that they use the same condition.